packages/@webex/plugin-cc/src/services/task/TaskManager.ts

```diff
@@ -0,0 +1,119 @@
+    this.webSocketManager = webSocketManager;
+    this.registerTaskListeners();
+    this.registerIncomingCallEvent();
+  }
+
+  private handleIncomingWebCall = (call: ICall) => {
+    if (this.currentTask) {
+      this.emit(TASK_EVENTS.TASK_INCOMING, this.currentTask);
+      // this.currentTask.webCallMap[this.currentTask.data.interactionId] = call.getCallId();
+    }
+    this.call = call;
+  };
+
+  public registerIncomingCallEvent() {
+    this.webCallingService.on(LINE_EVENTS.INCOMING_CALL, this.handleIncomingWebCall);
+  }
+
+  public unregisterIncomingCallEvent() {
+    this.webCallingService.off(LINE_EVENTS.INCOMING_CALL, this.handleIncomingWebCall);
+  }
+
+  private registerTaskListeners() {
+    this.webSocketManager.on('message', (event) => {
+      const payload = JSON.parse(event);
+      if (payload.data) {
+        switch (payload.data.type) {
+          case CC_EVENTS.AGENT_CONTACT_RESERVED:
+            this.currentTask = new Task(this.contact, this.webCallingService, payload.data);
+
+            this.taskCollection[payload.data.interactionId] = this.currentTask;
+            if (this.webCallingService.loginOption !== LoginOption.BROWSER) {
+              this.emit(TASK_EVENTS.TASK_INCOMING, this.currentTask);
+            } else if (this.call) {
+              this.emit(TASK_EVENTS.TASK_INCOMING, this.currentTask);
+              // this.currentTask.webCallMap[this.currentTask.data.interactionId] = this.call.getCallId();
+            }
+            break;
+          case CC_EVENTS.AGENT_CONTACT_ASSIGNED:
+            this.currentTask = this.currentTask.updateTaskData(payload.data);
+            this.emit(TASK_EVENTS.TASK_ASSIGNED, this.currentTask);
+            break;
+          case CC_EVENTS.CONTACT_ENDED:
+            this.emit(TASK_EVENTS.TASK_UNASSIGNED, this.currentTask);
+            if (this.webCallingService.loginOption === LoginOption.BROWSER) {
+              this.currentTask.unregisterWebCallListeners();
+              this.webCallingService.unregisterCallListeners();
+            }
+            break;
+          default:
+            break;
+        }
+      }
+    });
+  }
+
+  /**
+   * @param taskId - string.
+   */
+  public getTask = (taskId: string) => {
+    return this.taskCollection[taskId];
+  };
+
+  /**
+   *
+   */
+  public getAllTasks = (): Record<TaskId, ITask> => {
+    return this.taskCollection;
+  };
+
+  /**
+   * @param contact - Routing Contact layer. Talks to AQMReq layer to convert events to promises
+   * @param webCallingService - Webrtc Service Layer
+   * @param webSocketManager - Websocket Manager to maintain websocket connection and keepalives
+   */
+  public static getTaskManager = (
+    contact: ReturnType<typeof routingContact>,
+    webCallingService: WebCallingService,
+    webSocketManager: WebSocketManager
+  ): TaskManager => {
+    if (!this.taskManager) {
+      this.taskManager = new TaskManager(contact, webCallingService, webSocketManager);
```

Comment:
There isn't a test for this. We're always initializing the task manager before running any test.

Reply:
I tried to change this, but tests started to fail